### PR TITLE
fix pca ValueError

### PR DESCRIPTION
**Pull request by cxhernandez**
re: https://github.com/msmbuilder/osprey/issues/105

### CODE REVIEW

File: osprey/eval_scopes.py [hunk 1]
```diff
@@ -41,7 +41,8 @@ def estimator_in_module(mod):
             if ispkg:
                 result.update(import_all_estimators(mod))
             for kls in estimator_in_module(mod):
-                result[kls.__name__] = kls
+                if kls.__name__ not in result.keys():
```
Comment (cxhernandez) [inline osprey/eval_scopes.py]: @mpharrigan: let me know if you think this is a satisfactory fix. Another condition could be to only allow a module to replace another if it's from `msmbuilder`.

**Comment by mpharrigan** [comment]
Is there any reason to believe the first thing of a given name found is the one you want? You can imagine if the sklearn pca is found first.

Really, I think the fix is to manually enumerate all the things from msmbuilder we want to import, and if something's missing, someone will complain and it will be clear how to add it. This introspection stuff is "too clever"

**Comment by cxhernandez** [comment]
> This introspection stuff is "too clever"

Yeah, it's still unclear to me how it's finding the sklearn `PCA`

> manually enumerate all the things from msmbuilder we want to import

So make something like a [`FrozenDict`](http://stackoverflow.com/a/2704866) to replace `import_all_estimators`?

**Comment by cxhernandez** [comment]
Or maybe just a hard-coded `dict` and then add a flag in the `yaml` that allows users to add desired packages?

File: osprey/eval_scopes.py [hunk 2]
```diff
@@ -41,7 +42,7 @@ def estimator_in_module(mod):
             if ispkg:
                 result.update(import_all_estimators(mod))
             for kls in estimator_in_module(mod):
-                result[kls.__name__] = kls
+                result['.'.join([kls.__module__, kls.__name__])] = kls
```
Comment (cxhernandez) [inline osprey/eval_scopes.py]: this way nothing conflicts with the standard scope

**Comment by cxhernandez** [comment]
> Is there any reason to believe the first thing of a given name found is the one you want? You can imagine if the sklearn pca is found first.

How about something like:

```
if kls.__module__ == pkg:
    result[kls.__name__] = kls
```

This way it only adds things from the `pkg` it's searching for explicitly.

**Comment by mpharrigan** [comment]
Does that work?

**Comment by cxhernandez** [comment]
This works now :smile:

**Pull request merged**